vec.hpp

```diff
@@ -7,6 +7,16 @@ struct vec
     T v[N];
 };
 
+template <typename T, size_t N>
+std::ostream& operator<<(std::ostream& os, const vec<T, N>&vec) {
+	os << "{";
+	for(int i=0; i < N - 1; i++)
```

Comment:
Doesn't compile on my machine:

```console
vec.hpp:13:17: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
  for(int i=0; i < N - 1; i++)
                 ^
```

Please use `size_t` instead of `int`.